### ISSUE TITLE
Support UIAutomator tests of Flutter module in existing android app

### ISSUE THREAD

**Issue by growdai1y** [flutter/flutter#53383]
Hello.

I've been on investigation on flutter and made a decision that flutter is wonderful to cover legacy applications. 
However, when it comes to integration testing (or acceptance testing) with legacy applications, I've been failing to figure out how.

Here is my situation.

![image](https://user-images.githubusercontent.com/2580013/77709326-042afa80-700e-11ea-91c1-d14bcffab989.png)

I'd like to test "a new feature by flutter". But like I mentioned before, I failed to find a way.

Is it possible to run "flutter drive" on an existing android app that contains flutter module in it?

**Comment by jmagman** [comment]
\cc @xster

**Comment by xster** [comment]
cc @collinjackson 

Somewhat. You can't "flutter drive" (the Dart VM in your app might not even exist until a couple of screens later, the Dart VM can't drive your tests). 

The open source espresso now supports Flutter! See https://github.com/flutter/samples/pull/323/files for an example (thanks @RedBrogdon!)

For iOS, it's in progress. You can test some things via standard XCTest like https://cs.opensource.google/flutter/engine/+/master:testing/scenario_app/ios/Scenarios/ScenariosTests/AppLifecycleTests.m. You currently can't assert against the UI state inside Flutter like you could with Espresso but you can assert business logic via platform channel for now. 

https://github.com/flutter/flutter/issues/32987 tracking EarlGrey

**Comment by dudgus** [comment]
@xster Is there a way if I can guarantee the dart vm runs before getting into the flutter drvier test? 
I definitely guarantee that cause I do know when the android host app displays flutter views.

**Comment by xster** [comment]
Could you give us a bit more details? 

I'm getting the impression that there are 2 conflicting things. I understood the desire for UIAutomator driving blackbox testing. I'm not sure I understood the flutter drive part. How would you know there's a Flutter implementation of a module inside the blackbox? Do you have existing UIAutomator tests and are asking how to find an equivalent of new UiSelector().description("something") for a Flutter object?

**Comment by younghyunk** [comment]
> Could you give us a bit more details?
> 
> I'm getting the impression that there are 2 conflicting things. I understood the desire for UIAutomator driving blackbox testing. I'm not sure I understood the flutter drive part. How would you know there's a Flutter implementation of a module inside the blackbox? Do you have existing UIAutomator tests and are asking how to find an equivalent of new UiSelector().description("something") for a Flutter object?

You're right I'm looking for a way to access flutter objects with uiautomator.
